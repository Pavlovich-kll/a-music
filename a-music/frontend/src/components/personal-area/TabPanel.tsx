import Box from '@material-ui/core/Box';
import React from 'react';

interface TabPanelProps {
  children?: React.ReactNode;
  index: any;
  value: any;
}

export const TabPanel = (props: TabPanelProps) => {
  const { children, value, index, ...other } = props;

  return (
    <div
      className="tab"
      role="tabpanel"
      hidden={value !== index}
      id={`simple-tabpanel-${index}`}
      aria-labelledby={`simple-tab-${index}`}
      {...other}
    >
      {value === index && (
        <Box>
          <div>{children}</div>
        </Box>
      )}
    </div>
  );
};
