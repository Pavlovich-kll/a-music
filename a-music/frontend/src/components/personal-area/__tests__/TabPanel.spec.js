import React from "react";
import {TabPanel} from "../TabPanel";
import Enzyme from "enzyme";
import { mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";


Enzyme.configure({ adapter: new Adapter() });

let wrapper;
describe('tests TabPane components', () => {

    beforeEach(()=> {
        wrapper = mount(<TabPanel />)
    });

    it("render correctly component", () => {
        expect(wrapper).toMatchSnapshot();
    });
});