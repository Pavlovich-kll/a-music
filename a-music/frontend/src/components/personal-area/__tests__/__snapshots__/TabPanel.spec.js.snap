// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests TabPane components render correctly component 1`] = `
<TabPanel>
  <div
    aria-labelledby="simple-tab-undefined"
    className="tab"
    hidden={false}
    id="simple-tabpanel-undefined"
    role="tabpanel"
  >
    <Styled(MuiBox)>
      <div
        className="MuiBox-root MuiBox-root-2"
      >
        <WithStyles(ForwardRef(Typography))>
          <ForwardRef(Typography)
            classes={
              Object {
                "alignCenter": "MuiTypography-alignCenter",
                "alignJustify": "MuiTypography-alignJustify",
                "alignLeft": "MuiTypography-alignLeft",
                "alignRight": "MuiTypography-alignRight",
                "body1": "MuiTypography-body1",
                "body2": "MuiTypography-body2",
                "button": "MuiTypography-button",
                "caption": "MuiTypography-caption",
                "colorError": "MuiTypography-colorError",
                "colorInherit": "MuiTypography-colorInherit",
                "colorPrimary": "MuiTypography-colorPrimary",
                "colorSecondary": "MuiTypography-colorSecondary",
                "colorTextPrimary": "MuiTypography-colorTextPrimary",
                "colorTextSecondary": "MuiTypography-colorTextSecondary",
                "displayBlock": "MuiTypography-displayBlock",
                "displayInline": "MuiTypography-displayInline",
                "gutterBottom": "MuiTypography-gutterBottom",
                "h1": "MuiTypography-h1",
                "h2": "MuiTypography-h2",
                "h3": "MuiTypography-h3",
                "h4": "MuiTypography-h4",
                "h5": "MuiTypography-h5",
                "h6": "MuiTypography-h6",
                "noWrap": "MuiTypography-noWrap",
                "overline": "MuiTypography-overline",
                "paragraph": "MuiTypography-paragraph",
                "root": "MuiTypography-root",
                "srOnly": "MuiTypography-srOnly",
                "subtitle1": "MuiTypography-subtitle1",
                "subtitle2": "MuiTypography-subtitle2",
              }
            }
          >
            <p
              className="MuiTypography-root MuiTypography-body1"
            />
          </ForwardRef(Typography)>
        </WithStyles(ForwardRef(Typography))>
      </div>
    </Styled(MuiBox)>
  </div>
</TabPanel>
`;
